crates/executor/src/lib.rs

```diff
@@ -7,9 +7,27 @@
 
 //! Reth executor executes transaction in block of data.
 
+use async_trait::async_trait;
+use executor::ExecutionResult;
+use reth_interfaces::executor::Error;
+use reth_primitives::{Address, Block};
+
 pub mod config;
 pub mod eth_dao_fork;
 /// Executor
 pub mod executor;
 /// Wrapper around revm database and types
 pub mod revm_wrap;
+
+/// Takes block and executes it, returns result
+#[async_trait]
+pub trait BlockExecutor {
+    /// Execute block
+    /// if `signers` is some, it's length should be equal to block.body's length. Provide `signers`
+    /// if you already have it because recovering singer is too expensive.
+    fn execute(
+        &mut self,
+        block: &Block,
+        signers: Option<Vec<Address>>,
```

Comment:
If we do not end up removing this (see my above comments) we should at least change the name to `senders`

Reply:
I'm still learning and asking feedback if we could remove it or not. But I renamed it first. w/ b6a57b74400f7f8d9f626e851092534badbfe2ba, 3303ac649e56e43407be70011b19bb91956f3b69.